### DISCUSSION TITLE
Docker changie development image to help to contribute

### DISCUSSION THREAD

**Discussion by llaville** [miniscruff/changie#400] [Ideas] [answered]
Hello,

A docker image of changie with all tools (`go`, `hugo`, ...) pre-installed will be very helpfull to help to contribute.

I was in trouble to install on my plaform hugo-extended due to comment https://gohugo.io/getting-started/installing#debian-and-ubuntu, because I've Ubuntu 18..04 LTS

**Comment by miniscruff** [comment]
Are you expecting a dockerfile in the repo or just an example of what one would look like?

**Reply by llaville** [reply]
A concrete Dockerfile would be great !

**Reply by miniscruff** [reply]
Hmm ok, let me make an issue for it and maybe I can write something soon.

**Comment by miniscruff** [comment, marked as answer]
I was working on this today and got a basic dockerfile to work, combining this with a binding mount of your local directory should get you started. I am not quite sure I want to support this environment officially as it is not something I would use or even recommend.

```Dockerfile
FROM golang:1.18

WORKDIR /app

# Install tools
RUN go install golang.org/x/tools/cmd/goimports@latest &&
    go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.50.1 &&
    go install -tags extended github.com/gohugoio/hugo@latest
```

```shell
docker build -t changie:dev . -f dev.Dockerfile # or whatever name you give it
docker run --mount type=bind,source="$(pwd)",target=/app changie:dev make test # run unit tests
docker run --mount type=bind,source="$(pwd)",target=/app -p 1313:1313 -it changie:dev hugo serve -s docs --bind=0.0.0.0 # is the full command to run hugo
```

FYI @llaville

**Comment by sheldonhull** [comment]
If it would be useful I could contribute a little here with a devcontainer, unless it's desired to run completely locally. If locally with only dockerized commands I actually have Magefiles that are ready and been using for a while I could add for hugo serve, docker build etc. 

That means you drop into the repo and as long as you have go + mage (go install it) you can `mage serve`, `mage docker:build` etc. Be minimal effort for me to share that if you'd like.

Big fan of it and comes standard in almost all my projects now. Nice accelerator for contributing too. 

[one example](https://github.com/sheldonhull/digital-garden/blob/a7722e822838ef85b3d5b8e5d598bfd15a2d9f03/magefile.go#L66C10-L66C10)
